components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -375,6 +375,8 @@ impl DedicatedWorkerGlobalScope {
                 let scope = global.upcast::<WorkerGlobalScope>();
                 let global_scope = global.upcast::<GlobalScope>();
 
+                global_scope.set_https_state(current_global.get_https_state());
```

Comment:
I'm 99% confident that this doesn't build, since current_global can't be moved between threads. We need to get the https state value before spawning the new thread instead.

Reply:
Yes, fixed that. Strangely it did build. May be some caching... Also sometimes rustc crashes and builds fine after rerun.